### PR TITLE
New bootstrap 0504

### PR DESCRIPTION
**Pull request by alivedise**

### CODE REVIEW

**Comment by autolander** [comment]
Autolander could not find a bug number in your pull request title. All pull requests should be in the format of: Bug [number] - [description].

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=fb338a2c1d5588fcf0a2bff72b5411af99dde863

**Comment by autolander** [comment]
Autolander could not find a bug number in your pull request title. All pull requests should be in the format of: Bug [number] - [description].

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=820819b23fcb72bf83c2f43d3660d1f0887140de

**Comment by autolander** [comment]
Autolander could not find a bug number in your pull request title. All pull requests should be in the format of: Bug [number] - [description].

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=15fe75af64be58de5f23e90fd27badda8584b3b7

**Comment by autolander** [comment]
Autolander could not find a bug number in your pull request title. All pull requests should be in the format of: Bug [number] - [description].

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=248d3b7411aee9ca2a09cc921b1cb6b8f0589495

**Comment by autolander** [comment]
Autolander could not find a bug number in your pull request title. All pull requests should be in the format of: Bug [number] - [description].

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=c71c4734c2ac3e9631c10d7db3fcb9613f7f39c6

**Comment by autolander** [comment]
Autolander could not find a bug number in your pull request title. All pull requests should be in the format of: Bug [number] - [description].

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=40de533876c02538729875f1ba809bf32e6787d0

**Comment by autolander** [comment]
Autolander could not find a bug number in your pull request title. All pull requests should be in the format of: Bug [number] - [description].

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=f46a0d308a6f1d39db7facbaf07ecbffceb6b4a1

**Comment by autolander** [comment]
Autolander could not find a bug number in your pull request title. All pull requests should be in the format of: Bug [number] - [description].

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=ac3da3eefc2c5abbe80e28a5b89c549dfccc7ea8

**Comment by autolander** [comment]
Autolander could not find a bug number in your pull request title. All pull requests should be in the format of: Bug [number] - [description].

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=5f92061cf422c77f52491f90deffe6ef5727954f

**Comment by autolander** [comment]
Autolander could not find a bug number in your pull request title. All pull requests should be in the format of: Bug [number] - [description].

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=fdb30e636142eb3bdde1e98dcea1928291ed4813

**Comment by autolander** [comment]
Autolander could not find a bug number in your pull request title. All pull requests should be in the format of: Bug [number] - [description].

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=79287b93beba3f987509d4f99dfbbc60a7ab259c

**Comment by autolander** [comment]
Autolander could not find a bug number in your pull request title. All pull requests should be in the format of: Bug [number] - [description].

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=a62c56d27010ac5096b1f71bc6d72a2927734556

**Comment by autolander** [comment]
Autolander could not find a bug number in your pull request title. All pull requests should be in the format of: Bug [number] - [description].

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=3501bc7f161e95c32ba8ab8a1251842f068ae4a7

**Comment by autolander** [comment]
Autolander could not find a bug number in your pull request title. All pull requests should be in the format of: Bug [number] - [description].

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=9e2c60339c58404191e084cfb41bcaba7ff0b196

File: apps/system/js/app.js [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+        // but it may miss the mozChromeEvent to tell us application
+        // is ready to query.
+        applications.waitForReady(),
+        window.launcher.start()
+      ]).then(() => {
+        window.core = BaseModule.instantiate('Core');
+        return window.core.start().then(() => {
+          // To let integration test know we are ready to test.
+          document.body.setAttribute('ready-state', 'fullyLoaded');
+          window.performance.mark('fullyLoaded');
```
Comment (timdream) [inline apps/system/js/app.js]: To avoid nesting you can move `then(...) from L72-75 to the end at L77.

**Comment by autolander** [comment]
Autolander could not find a bug number in your pull request title. All pull requests should be in the format of: Bug [number] - [description].

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=48e58b937b4ee72d45d02d2fff5e3bf0b7e7d2d2

File: apps/system/js/wallpaper_manager.js [hunk 2]
```diff
@@ -48,43 +48,60 @@
...
-      query.onsuccess = function() {
-        var wallpaper = query.result[WALLPAPER_KEY];
+    initializeWallpaper: function(wallpaper, valid) {
+      return new Promise((resolve, reject) => {
+        if (this.wallpaperInitialized) {
+          reject();
+          return;
+        }
+        this.wallpaperInitialized = true;
+        this.initPromise = resolve;
```
Comment (timdream) [inline apps/system/js/wallpaper_manager.js]: `initPromise` -> `_initPromiseResolver`

**Comment by autolander** [comment]
Autolander could not find a bug number in your pull request title. All pull requests should be in the format of: Bug [number] - [description].

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=041c66cc3c59b55dd73e7f2df6310a71407a2be7

**Comment by autolander** [comment]
The pull request could not be integrated in as it contained a commit which was missing a bug number. Please ensure all commits contain bug numbers, and are in the format of: Bug [number] - [description]

**Pull request closed without merging**